core/src/main/java/it/pagopa/selfcare/mscore/core/strategy/CreateInstitutionStrategyIpa.java

```diff
@@ -0,0 +1,197 @@
+import it.pagopa.selfcare.mscore.model.institution.*;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.lang3.StringUtils;
+import org.springframework.stereotype.Component;
+
+import java.time.OffsetDateTime;
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+
+import static it.pagopa.selfcare.mscore.constant.GenericError.CREATE_INSTITUTION_ERROR;
+
+@Slf4j
+@Component
+public class CreateInstitutionStrategyIpa implements CreateInstitutionStrategy {
+
+    private final PartyRegistryProxyConnector partyRegistryProxyConnector;
+
+    private final InstitutionConnector institutionConnector;
+
+    private final InstitutionMapper institutionMapper;
+
+    public CreateInstitutionStrategyIpa(PartyRegistryProxyConnector partyRegistryProxyConnector,
+                                        InstitutionConnector institutionConnector,
+                                        InstitutionMapper institutionMapper) {
+        this.partyRegistryProxyConnector = partyRegistryProxyConnector;
+        this.institutionConnector = institutionConnector;
+        this.institutionMapper = institutionMapper;
+    }
+
+
+    @Override
+    public Institution createInstitution(CreateInstitutionStrategyInput strategyInput) {
+
+        final InstitutionPaSubunitType subunitType = strategyInput.getSubunitType();
+
+        Institution institutionToReturn;
+        Optional<Institution> opt = institutionConnector.findByExternalId(strategyInput.getTaxCode());
+        if(opt.isEmpty()){
+            try {
+                Institution institutionEC = getInstitutionEC(strategyInput.getTaxCode());
+                institutionToReturn = institutionConnector.save(institutionEC);
+            } catch (Exception e) {
+                throw new MsCoreException(CREATE_INSTITUTION_ERROR.getMessage(), CREATE_INSTITUTION_ERROR.getCode());
+            }
+        } else {
+           if(Objects.isNull(subunitType) || InstitutionPaSubunitType.EC.equals(subunitType)){
+               throw new ResourceConflictException(String
+                       .format(CustomError.CREATE_INSTITUTION_CONFLICT.getMessage(), strategyInput.getTaxCode()),
+                       CustomError.CREATE_INSTITUTION_CONFLICT.getCode());
+           }
+            institutionToReturn = opt.get();
+        }
+
+        checkIfAlreadyExistsByTaxCodeAndSubunitCode(strategyInput.getTaxCode(), strategyInput.getSubunitCode());
+
+        Institution institution;
+        if (InstitutionPaSubunitType.AOO.equals(subunitType)) {
+            institution = mappingToInstitutionIPAAoo(strategyInput, institutionToReturn);
+        } else if(InstitutionPaSubunitType.UO.equals(subunitType)) {
+            institution = mappingToInstitutionIPAUo(strategyInput, institutionToReturn);
+        }else{
+            log.info("createInstitution :: unsupported subunitType {}", subunitType);
+            return institutionToReturn;
+        }
```

Comment:
you might refactoring the method to avoid calling twice:

  InstitutionProxyInfo institutionProxyInfo = partyRegistryProxyConnector.getInstitutionById(taxCode);
        CategoryProxyInfo categoryProxyInfo = partyRegistryProxyConnector.getCategory(institutionProxyInfo.getOrigin(), institutionProxyInfo.getCategory());

if an institution is AOO/UO and EC is not present, getInstitutionEc and mappingToInstitution_ will call the snippet above....